### PR TITLE
fix to decrement open connections count on connect exception

### PR DESCRIPTION
**Pull request by V0idmain**
Fixes the issue where the pool connection count in incremented even if their is an error while creating the connection. Based on an earlier PR that was never merged.

### CODE REVIEW

**Comment by benjsto** [comment]
This code looks good to me but I think we'll obv want to test it in stage with an existing service (like flexpro_api) to make sure it a) solves the problem we're trying to solve and b) doesn't create any other weird behavior.

**Comment by V0idmain** [comment]
@benjsto agreed. Can you pull it in to flexpro, push it and we can test?

**Comment by benjsto** [comment]
I can do that. Won't get to that until Monday most likely..

File: tornado_mysql/pools.py [hunk 1]
```diff
@@ -72,13 +72,20 @@ def _get_conn(self):
...
+            fut.add_done_callback(self._on_connect)  # self._opened_conns -=1 on exception
+            return fut
 
         # Wait to other connection is released.
         fut = Future()
         self._waitings.append(fut)
         return fut
 
+    def _on_connect(self, fut):
+        logging.info('error')
```
Comment (dangnvang) [inline tornado_mysql/pools.py]: Looks like this should be inside the if block below?

**Comment by dangnvang** [comment]
@benjsto: I'll take a shot at a PR for flexpro

**Comment by benjsto** [comment]
Thanks Dang!

File: tornado_mysql/pools.py (hunk 1 above)
Comment (V0idmain) [reply tornado_mysql/pools.py]: thanks @dangnvang will just remove it.

**Pull request merged**